tests/test_preprocessing.py

```diff
@@ -247,28 +278,32 @@ def test_sample_copy_backed_error(tmp_path):
 
 
 @pytest.mark.parametrize("array_type", ARRAY_TYPES)
-@pytest.mark.parametrize(
-    "zero_center", [True, False], ids=["zero_center", "no_zero_center"]
-)
 @pytest.mark.parametrize("max_value", [None, 1.0], ids=["no_clip", "clip"])
 def test_scale_matrix_types(array_type, zero_center, max_value):
     adata = pbmc68k_reduced()
     adata.X = adata.raw.X
     adata_casted = adata.copy()
     adata_casted.X = array_type(adata_casted.raw.X)
-    sc.pp.scale(adata, zero_center=zero_center, max_value=max_value)
-    with maybe_dask_process_context():
+    warn_ctx = pytest.warns(UserWarning, match=r"zero-centering.*densifies")
+    with warn_ctx if zero_center else nullcontext():
+        sc.pp.scale(adata, zero_center=zero_center, max_value=max_value)
+    with (
+        (
+            warn_ctx
+            if zero_center
+            and any(pat in array_type.__name__ for pat in ("sparse", "csc", "csr"))
```

Comment:
I made this more elegant for `count_matrix_format`, but to really fix that, we need fast-array-utils’ testing utils.